cmd/kubeadm/app/phases/addons/manifests.go

```diff
@@ -122,8 +118,6 @@ spec:
         k8s-app: kube-dns
       annotations:
         scheduler.alpha.kubernetes.io/critical-pod: ''
-        # TODO: Move this to the beta tolerations field below as soon as the Tolerations field exists in PodSpec
-        scheduler.alpha.kubernetes.io/tolerations: '[{"key":"CriticalAddonsOnly", "operator":"Exists"}, {"key":"dedicated","value":"master","effect":"NoSchedule"}]'
```

Comment:
You seem to have forgotten to carry over the "CriticalAddonsOnly Exists" toleration.